File: support_tools/swigwin-3.0.12/Examples/test-suite/octave/dynamic_cast_runme.m
# do not dump Octave core
if exist("crash_dumps_octave_core", "builtin")
  crash_dumps_octave_core(0);
endif

dynamic_cast

f = dynamic_cast.Foo();
b = dynamic_cast.Bar();

x = f.blah();
y = b.blah();

a = dynamic_cast.do_test(y);
if (!strcmp(a,"Bar::test"))
  error("Failed!!")
endif

    

File: support_tools/swigwin-3.0.12/Examples/test-suite/java/dynamic_cast_runme.java

import dynamic_cast.*;

public class dynamic_cast_runme {
  static {
    try {
        System.loadLibrary("dynamic_cast");
    } catch (UnsatisfiedLinkError e) {
      System.err.println("Native code library failed to load. See the chapter on Dynamic Linking Problems in the SWIG Java documentation for help.\n" + e);
      System.exit(1);
    }
  }

  public static void main(String argv[]) 
  {
    Foo f = new Foo();
    Bar b = new Bar();

    Foo x = f.blah();
    Foo y = b.blah();

    // Note it is possible to downcast y with a Java cast.
    String a = dynamic_cast.do_test((Bar)y);
    if (!a.equals("Bar::test")) {
        throw new RuntimeException("Failed!");
    }
  }
}
